packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx

```diff
@@ -0,0 +1,272 @@
+ */
+@Component({
+  tag: 'atomic-answerapi-generated-answer',
+  styleUrl: 'atomic-answerapi-generated-answer.pcss',
+  shadow: true,
+})
+export class AtomicAnswerApiGeneratedAnswer implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  public generatedAnswer!: GeneratedAnswer;
+  public searchStatus!: SearchStatus;
+  private resizeObserver?: ResizeObserver;
+
+  @BindStateToController('generatedAnswer', {
+    onUpdateCallbackMethod: 'onGeneratedAnswerStateUpdate',
+  })
+  @State()
+  private generatedAnswerState!: GeneratedAnswerState;
+
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+
+  @State()
+  public error!: Error;
+
+  @Element() private host!: HTMLElement;
+
+  @State()
+  copied = false;
+
+  @State()
+  copyError = false;
+
+  @Prop({reflect: true}) answerConfigurationId?: string;
+
+  /**
+   * The answer style to apply when the component first loads.
+   * Options:
+   *   - `default`: Generate the answer without specific formatting instructions.
+   *   - `bullet`: Generate the answer as a bulleted list.
+   *   - `step`: Generate the answer as step-by-step instructions.
+   *   - `concise`: Generate the answer as briefly as possible.
+   */
+  @Prop() answerStyle: GeneratedAnswerStyle = 'default';
+
+  /**
+   * Whether to render a toggle button that lets the user hide or show the answer.
+   * @default false
+   */
+  @Prop() withToggle?: boolean;
+
+  /**
+   * Whether to allow the answer to be collapsed when the text is taller than 250px.
+   * @default false
+   */
+  @Prop() collapsible?: boolean;
+
+  @AriaLiveRegion('generated-answer')
+  protected ariaMessage!: string;
+
+  private generatedAnswerCommon!: GeneratedAnswerCommon;
+  private fullAnswerHeight?: number;
+  private maxCollapsedHeight = 250;
+
+  public initialize() {
+    this.generatedAnswerCommon = new GeneratedAnswerCommon({
+      host: this.host,
+      withToggle: this.withToggle,
+      collapsible: this.collapsible,
+      getGeneratedAnswer: () => this.generatedAnswer,
+      getGeneratedAnswerState: () => this.generatedAnswerState,
+      getSearchStatusState: () => this.searchStatusState,
+      getBindings: () => this.bindings,
+      getCopied: () => this.copied,
+      setCopied: this.setCopied,
+      getCopyError: () => this.copyError,
+      setCopyError: this.setCopyError,
+      setAriaMessage: this.setAriaMessage,
+      buildInteractiveCitation: (props) =>
+        buildInteractiveCitation(this.bindings.engine, props),
+    });
+    this.generatedAnswer = buildGeneratedAnswer(this.bindings.engine, {
+      initialState: {
+        isVisible: this.generatedAnswerCommon.data.isVisible,
+        responseFormat: {
+          answerStyle: this.answerStyle,
+          contentFormat: ['text/markdown', 'text/plain'],
+        },
+      },
+      answerConfigurationId: this.answerConfigurationId,
```

Comment:
I find it somewhat redundant to duplicate the entire `atomic-generated-answer` component just to add this single line. What was the reasoning behind this ? Would it not have been simpler to add an option to the existing atomic-`atomic-generated-answer` component?

Reply:
After a dm discussion, my opinion still stands. 
We currently have a public headless controller diverging to 2 APIs via a prop. Why shouldn't we do the same thing on the atomic side and reuse the `AtomicGenratedAnswer` ?

Reply:
@tedre191, that ain't what we chatted about. :suspect:

Reply:
Agreed, if this is the only change then I don't see the need to duplicate the entire component.

We can instead add an optional prop on the `atomic-generated-answer` component and do:

```typescript
      ...(this.answerConfigurationId && {
        answerConfigurationId: this.answerConfigurationId,
      }),
```

here